api/v1/models/controller_status.go

```diff
@@ -169,7 +169,7 @@ func (m *ControllerStatusConfiguration) UnmarshalBinary(b []byte) error {
 type ControllerStatusStatus struct {
 
 	// Number of consecutive errors since last success
-	ConsecutiveFailureCount int64 `json:"consecutive-failure-count,omitempty"`
+	ConsecutiveFailureCount int64 `json:"consecutive-failure-count"`
```

Comment:
Are you sure we need this? I ran the jsonpath that you use from `master` and I don't have any problems with it
```
$ cilium status -o jsonpath='{range .controllers[*]}{.name}{"="}{.status.consecutive-failure-count}{"\n"}{end}'
kvstore-lease-keepalive=0
sync-identity-to-k8s-pod (5594)=0
sync-IPv4-identity-mapping (60221)=0
sync-policymap-51065=0
kvstore-sync-store-cilium/state/nodes/v1=0
sync-to-k8s-ciliumendpoint (60221)=0
sync-to-k8s-ciliumendpoint (60266)=0
sync-IPv4-identity-mapping (51065)=0
sync-to-k8s-ciliumendpoint-gc (k8s2)=0
kvstore-etcd-session-renew=0
sync-IPv4-identity-mapping (54070)=0
sync-identity-to-k8s-pod (60221)=0
sync-policymap-60221=0
lxcmap-bpf-host-sync=0
sync-to-k8s-ciliumendpoint (5594)=0
sync-IPv4-identity-mapping (5594)=0
sync-identity-to-k8s-pod (60266)=0
sync-identity-to-k8s-pod (51065)=0
resolve-identity-51065=0
sync-policymap-60266=0
sync-to-k8s-ciliumendpoint (51065)=0
sync-identity-to-k8s-pod (54070)=0
sync-to-k8s-ciliumendpoint (54070)=0
sync-IPv6-identity-mapping (54070)=0
sync-IPv6-identity-mapping (5594)=0
sync-IPv4-identity-mapping (60266)=0
sync-IPv6-identity-mapping (60266)=0
sync-IPv6-identity-mapping (60221)=0
cilium-health-ep=0
metricsmap-bpf-prom-sync=0
ipcache-bpf-garbage-collection=0
sync-policymap-54070=0
sync-policymap-5594=0
sync-IPv6-identity-mapping (51065)=0
```

Reply:
I think that it's useful information to have. I can use the jsonpath and checking if it's empty, but I don't think that it's a good approach. 

Furthermore, I don't think that it's good to expose data in certain cases and not in others.

Reply:
@eloycoto I was saying, without your changes I can get the `0` with jsonpath. But with `-o json` `consecutive-failure-count` does not show up.

the omitempty avoids wasting resources while doing marshalling and unmarshalling "zero" values.